search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/GenerativeQAResponseProcessor.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2023 Aryn
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.opensearch.searchpipelines.questionanswering.generative;
+
+import com.google.common.base.Preconditions;
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.search.TotalHits;
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.action.search.SearchResponseSections;
+import org.opensearch.client.Client;
+import org.opensearch.common.document.DocumentField;
+//import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.ingest.ConfigurationUtils;
+//import org.opensearch.neuralsearch.query.NeuralQueryBuilder;
+import org.opensearch.search.SearchHit;
+import org.opensearch.search.SearchHits;
+import org.opensearch.search.pipeline.AbstractProcessor;
+import org.opensearch.search.pipeline.Processor;
+import org.opensearch.search.pipeline.SearchResponseProcessor;
+import org.opensearch.searchpipelines.questionanswering.generative.llm.ChatCompletionOutput;
+import org.opensearch.searchpipelines.questionanswering.generative.llm.Llm;
+import org.opensearch.searchpipelines.questionanswering.generative.llm.LlmIOUtil;
+import org.opensearch.searchpipelines.questionanswering.generative.llm.ModelLocator;
+
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Defines the response processor for generative QA search pipelines.
+ *
+ */
+public class GenerativeQAResponseProcessor extends AbstractProcessor implements SearchResponseProcessor {
+
+    private static final Logger logger = LogManager.getLogger();
+    public static final String TYPE = "generative_qa";
+    public static final String CONFIG_NAME_OPENSEARCH_MODEL_ID = "opensearch_model_id";
+    public static final String CONFIG_NAME_LLM_MODEL = "llm_model";
+    public static final String CONFIG_NAME_CONTEXT_FIELD = "context_field";
+
+    // TODO Add "interaction_count".  This is how far back in chat history we want to go back when calling LLM.
+
+    private final String openSearchModelId;
+    private final String llmModel;
```

Comment:
llmModel sounds like: largelanguagemodelModel. 

May be we can change to: `llmName`?

Reply:
This model information exists in model config when registering with ml-commons, right? So we need model id here?

Reply:
No, check out the OpenAI API.  You can choose a difference model on each request.

Reply:
Actually, even the Connector template supports that.